src/components/button/button.test.jsx

```diff
@@ -14,136 +15,136 @@ describe('Button', () => {
     });
 
     it('can be "secondary"', () => {
-      const tree = renderer.create((
+      render((
         <Button color="secondary">
           Hello world!
         </Button>
-      )).toJSON();
-      expect(tree.props.className).toEqual('secondary');
+      ));
+      expect(screen.getByText('Hello world!')).toHaveClass('secondary');
     });
   });
 
   describe('disabled API', () => {
     it('can be disabled', () => {
-      const tree = renderer.create((
+      render((
         <Button disabled={true}>
           Hello world!
         </Button>
-      )).toJSON();
-      expect(tree.props.disabled).toEqual(true);
+      ));
+      expect(screen.getByText('Hello world!')).toBeDisabled();
     });
 
     it('can be enabled', () => {
-      const tree = renderer.create((
+      render((
         <Button disabled={false}>
           Hello world!
         </Button>
-      )).toJSON();
-      expect(tree.props.disabled).toEqual(false);
+      ));
+      expect(screen.getByText('Hello world!')).toBeEnabled();
     });
   });
 
   describe('name API', () => {
     it('sets the name attribute', () => {
-      const tree = renderer.create((
+      render((
         <Button name="test-name">
           Hello world!
         </Button>
-      )).toJSON();
-      expect(tree.props.name).toEqual('test-name');
+      ));
+      expect(screen.getByText('Hello world!')).toHaveAttribute('name', 'test-name');
     });
   });
 
   describe('id API', () => {
     it('sets the id attribute', () => {
-      const tree = renderer.create((
+      render((
         <Button id="test-id">
           Hello world!
         </Button>
-      )).toJSON();
-      expect(tree.props.id).toEqual('test-id');
+      ));
+      expect(screen.getByText('Hello world!')).toHaveAttribute('id', 'test-id');
     });
   });
 
   describe('onClick API', () => {
     it('sets the onclick handler', () => {
       const onClickSpy = jest.fn();
-      const tree = renderer.create((
+      render((
         <Button onClick={onClickSpy}>
           Hello world!
         </Button>
-      )).toJSON();
-      tree.props.onClick();
+      ));
+      fireEvent.click(screen.getByText('Hello world!'));
       expect(onClickSpy.mock.calls.length).toEqual(1);
     });
   });
 
   describe('type API', () => {
     it('can be "button"', () => {
-      const tree = renderer.create((
+      render((
         <Button type="button">
           Hello world!
         </Button>
-      )).toJSON();
-      expect(tree.props.type).toEqual('button');
+      ));
+      expect(screen.getByText('Hello world!')).toHaveAttribute('type', 'button');
     });
 
     it('can be "reset"', () => {
-      const tree = renderer.create((
+      render((
         <Button type="reset">
           Hello world!
         </Button>
-      )).toJSON();
-      expect(tree.props.type).toEqual('reset');
+      ));
+      expect(screen.getByText('Hello world!')).toHaveAttribute('type', 'reset');
```

Comment:
I think generally speaking the `testing-library` wants to promote things a user can observe.

A user can't observe the type of the button, but they can observe its behavior. What does a `reset` button actually do? Does it have a specific class? Does it do something different on click?

Reply:
The `submit` and `reset` types are related to `<form>`s. If we were to add a form then we could test the side effects. I think that would be more of an integration test and it would be more applicable when using the component in a host application. 

In our world, we would have a test in our Mavenlink application repository which asserts that the actual form is being reset when clicking the button. 

I think this might be a subtle difference between `@testing-library/react` and `react-test-renderer`. We want as a library to test the direct API changes so we're perhaps going against the grain in our test files.